src/mlpack/methods/sparse_svm/sparse_svm.hpp

```diff
@@ -0,0 +1,104 @@
+/**
+ * @file sparse_svm.hpp
+ * @author Ayush Chamoli
+ *
+ * An implementation of Sparse SVM.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_SPARSE_SVM_SPARSE_SVM_HPP
+#define MLPACK_METHODS_SPARSE_SVM_SPARSE_SVM_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <ensmallen.hpp>
+
+#include "sparse_svm_function.hpp"
+
+namespace mlpack {
+namespace svm {
+
+class SparseSVM
+{
+ public:
+  /**
+   * Construct the SparseSVM class with the provided data and labels.
+   * This will train the model.
+   *
+   * @tparam OptimizerType Desired differentiable separable optimizer
+   * @param data Input training features. Each column associate with one sample
+   * @param labels Labels associated with the feature data.
+   * @param optimizer Desired optimizer.
+   */
+  template <typename OptimizerType = ens::ParallelSGD<>>
+  SparseSVM(const arma::mat& data,
+            const arma::Row<size_t>& labels,
+            OptimizerType optimizer = OptimizerType());
+
+  /**
+   * Classify the given points, returning  predicted class label ({-1, 1})
```

Comment:
`size_t` actually won't hold `-1` well, so we should make the class labels `0` or `1`.  You might need to rework the implementation for that a little bit too.

Reply:
I think I missed this and probably didn't notice it since I haven't made any tests on this class. But thanks for pointing this out, I will update the way it works around labels `0` and `1` rather than `-1` and `1`.